dask/dataframe/io/tests/test_io.py

```diff
@@ -514,6 +514,28 @@ def test_from_delayed():
     assert str(e.value).startswith('Metadata mismatch found in `from_delayed`')
 
 
+def test_from_delayed_misordered_meta():
+    df = pd.DataFrame(
+        columns=['(1)', '(2)', 'date', 'ent', 'val'],
+        data=[range(i * 5, i * 5 + 5) for i in range(3)],
+        index=range(3)
+    )
+
+    # meta with different order for columns
+    misordered_meta = pd.DataFrame(
+        columns=['date', 'ent', 'val', '(1)', '(2)'],
+        data=[range(5)]
+    )
+
+    ddf = dd.from_delayed([delayed(lambda: df)()], meta=misordered_meta)
+
+    with pytest.raises(ValueError) as info:
+        #produces dataframe which does not match meta
+        ddf.reset_index().compute()
+    msg = 'Meta columns do not match dataframe.'
```

Comment:
This error messages is clear to me, but may not be clear to a naive user.  Perhaps "The columns in the computed data do not match the columns in the provided metadata" or something similar?  I'm anticipating users asking what a "meta column" is :)

Reply:
@mrocklin thanks for the suggestion. Your message makes way more sense! Have changed it now.

Reply:
Would be much more informative to raise ValueError("The columns in the provided metadata do not match the columns in the computed data: {}".format(df.columns))